### PR TITLE
debian: Improve packaging

### PR DESCRIPTION
**Pull request by bdrung**
See commit message of the individual commits.

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -7,6 +7,10 @@
 # sensible basis of options (eg no -O3, including -g). Debian specific options
 # come from CFLAGS as usual.
 
+override_dh_auto_clean:
```
Comment (jgunthorpe) [inline debian/rules]: This new stanza ended up between two comment paragraphs that were intended to stay together, can you adjust it?

There is probably no reason to call ninja clean, rm -rf build-deb is all that is needed.

**Comment by bdrung** [comment]
Yes, I moved the stanza and use `rm -rf build-deb` for cleaning.

File: debian/control [hunk 2]
```diff
@@ -6,11 +6,9 @@ Build-Depends: cmake (>= 2.8.11),
                debhelper (>= 9),
                dh-systemd,
                dpkg-dev (>= 1.17),
-               gcc,
                libnl-3-dev,
                libnl-route-3-dev,
                libudev-dev,
-               make,
                ninja-build,
```
Comment (rleon) [inline debian/control]: Benjamin,
I have following commit in our review system.
Is it correct one? If yes, can you squash it?

commit 0e6099884e72d1a6a35238884353778d0083fe40
Author: Talat Batheesh <talatb@mellanox.com>
Date:   Mon Dec 26 10:35:16 2016 +0200

    debian/control: Unmet build dependencies ninja
    
    Fixes  the following issues
    
        * debuild clean issue
        * dpkg-checkbuilddeps: error: Unmet build dependencies: ninja
    
    Signed-off-by: Talat Batheesh <talatb@mellanox.com>

diff --git a/debian/control b/debian/control
index 3b04ac9..367c30f 100644
--- a/debian/control
+++ b/debian/control
@@ -12,7 +12,7 @@ Build-Depends: build-essential,
               libnl-3-dev,
               libnl-route-3-dev,
               make,
-              ninja,
+              ninja-build,
               pkg-config,
               valgrind
 Standards-Version: 3.9.8
~

Comment (bdrung) [inline debian/control]: Is that review system publicly available?

Assuming that I correctly read the ill-formatted pasted commit (please use block format the next time), Talat's commit is identical to my commit [debian: Correct name of ninja-build dependency](https://github.com/linux-rdma/rdma-core/pull/54/commits/0c142bcd127901faff3b27d6c4a94452567c4c4f) which is the second commit of this merge proposal.

File: debian/rules (hunk 1 above)
Comment (bdrung) [inline debian/rules]: Done.

File: debian/control (hunk 2 above)
Comment (rleon) [inline debian/control]: Yes, it looks the same, sorry for the format.

**Pull request merged**